bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/ModelController.java

```diff
@@ -159,6 +170,7 @@ public void update(final Context ctx, final String modeluri) {
                } catch (EncodingException e) {
                   handleEncodingError(ctx, e);
                }
+               this.modelRepository.validate(modeluri);
```

Comment:
undo and redo make sense for sure. I will add them. 
However, for save and when the model is manually validated i am not sure. You could argue, that when the model is saved, the model is not changing therefore the validation result will not be different than before. 
So broadcasting the validation result while saving/manually validating will act like a refresh, that may be able to solve some problems that happened before, but if nothing went wrong it will resend the same results to all the clients. 
Which way would you prefer?

Reply:
Yes you're totally right. I just took a deeper look at the validation behavior in the tree model editor in Eclipse for comparison:
- If live validation is activated, there is no need to send an additional validation update on saving.
- If live validation is deactivated, the client/user is responsible on its own to trigger validation anyway. In Eclipse it is also not triggered automatically on `save`, so therefore it's fine that way I would say :)